examples/mint-repl/mint-repl.rs

```diff
@@ -0,0 +1,1110 @@
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ReissueRequestOwned {
+    inner: ReissueRequest,
+    inputs_owners: HashMap<Hash, PublicKeySet>,
+    outputs_owners: HashMap<Hash, PublicKeySet>,
+}
+
+/// This type is just for serializing HashMap<Hash, <HashMap<usize, SignatureShare>>
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+struct SignatureSharesMap(HashMap<Hash, HashMap<usize, SignatureShare>>);
+
+/// program entry point and interactive command handler.
+fn main() -> Result<()> {
+    print_logo();
+    println!("Type 'help' to get started.\n");
+
+    let mut mintinfo: MintInfo = mk_new_random_mint(0, 1000)?;
+
+    let mut rl = Editor::<()>::new();
+    rl.set_auto_add_history(true);
+    'outer: loop {
+        match rl.readline(">> ") {
+            Ok(line) => {
+                let mut args = line.trim().split_whitespace();
+                let cmd = if let Some(cmd) = args.next() {
+                    cmd
+                } else {
+                    continue 'outer;
+                };
+                let result = match cmd {
+                    "newmint" => {
+                        mintinfo = newmint()?;
+                        Ok(())
+                    }
+                    "mintinfo" => print_mintinfo_human(&mintinfo),
+                    "prepare_tx" => prepare_tx(),
+                    "sign_tx" => sign_tx(),
+                    "prepare_reissue" => prepare_reissue(),
+                    "reissue" => reissue(&mut mintinfo),
+                    "reissue_ez" => reissue_ez(&mut mintinfo),
+                    "validate" => validate(&mintinfo),
+                    "newkey" => newkey(),
+                    "decode" => decode_input(),
+                    "quit" | "exit" => break 'outer,
+                    "help" => {
+                        println!(
+                            "\nCommands:\n  Mint:    [mintinfo, newmint, reissue]\n  Client:  [newkey, prepare_tx, sign_tx, prepare_reissue, reissue_ez, decode, validate]\n  General: [exit, help]\n"
+                        );
+                        Ok(())
+                    }
+                    _ => Err(anyhow!("Unknown command")),
+                };
+                if let Err(msg) = result {
+                    println!("Error: {}", msg);
+                }
+            }
+            Err(ReadlineError::Eof) | Err(ReadlineError::Interrupted) => break 'outer,
+            Err(e) => {
+                println!("Error reading line: {}", e);
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// handles newmint command to generate a new mint with N nodes.
+fn newmint() -> Result<MintInfo> {
+    let confirm = readline_prompt(
+        "\nThis will erase existing Mint and transactions.  Are you sure? [y/n]: ",
+    )?;
+    if confirm != "y" {
+        return Err(anyhow!("newmint operation cancelled"));
+    }
+
+    let amount = loop {
+        let amount: u64 = readline_prompt("\nTotal Money Supply Amount: ")?.parse()?;
```

Comment:
```suggestion
        let amount: u64 = readline_prompt("\nTotal Money Supply Amount (> 0): ")?.parse()?;
```

Reply:
I was thinking maybe I should just get rid of the > 0 check, and let sn_dbc validation logic handle it.  Especially in light of recent discussions that issuing 0 amounts Dbcs is presently possible.

Reply:
Yeah, that sounds better actually. Don't shield the DBC validation logic

Reply:
on second thought, I've changed it to a warning/prompt, and let user proceed or re-enter.